### DISCUSSION TITLE
Problème DEL dans metric dateUtil_serv.py

### DISCUSSION THREAD

**Discussion by elbourymohammed** [danymat/INSAnonym-utils#97] [Q&A, Bug report & Support]
Bonjour , 

J'écris ce message pour vous faire part que la version du metric dateUtil_serv.py , calcule toutes les lignes de la base de données même si il y'a des DEL au niveau de l'ID , et par conséquent elle ne les ignore pas.

Je vous informe par la même occasion que j'ai la version la plus récente de la metric dateUtil_serv.py.

J'aimerai bien avoir un retour s'il vous plaît à propos de cette problématique 

Merci d'avance

**Comment by ceichler** [comment]
Bonjour,

Cela semble bel et bien marcher, par contre j'imagine que vous n'appelez pas bien le script; le premier argument du script doit être la base de données anonyme et le second la base de données initiale.

Ainsi, étant données les DBs [truc.csv](https://github.com/danymat/INSAnonym-utils/files/7603402/truc.csv), [truca.csv](https://github.com/danymat/INSAnonym-utils/files/7603403/truca.csv) (une lignée trashée en date +3j) et [trucDEL.csv](https://github.com/danymat/INSAnonym-utils/files/7603404/trucDEL.csv) (même chose que truca mais la ligne trashée est DEL) 


:

- dateUtil truca truc donne 0,66 (2lignes sur trois parfaites, une trashée donnant un score nul)
- dateUtil trucDEL truc donne 1 (2 lignes parfaites, une ignorée car DEL)

Si cela vous perturbe vous pouvez inverser les lignes 49 et 50 du script:
   
    parser.add_argument("anonymized", help="Anonymized Dataframe filename")
    parser.add_argument("original", help="Original Dataframe filename")

en


```

    parser.add_argument("original", help="Original Dataframe filename")
    parser.add_argument("anonymized", help="Anonymized Dataframe filename")
```

**Reply by ceichler** [reply]
Par ailleurs ce comportement n'était pas le bon, les lignes DEL doivent bel et bien être pénalisées, comme c'était fait sur le serveur. Le script correct a été posté sur la discussion https://github.com/danymat/INSAnonym-utils/discussions/98#discussion-3706942

**Comment by beng-git** [comment]
Bonjour,

Pour préciser ce que fait la fonction meet sur le serveur : elle calcule le taux de recouvrement entre le top 10% des cellules les plus visités (avec une précision de 2 "quartier") dans le jeu de données initial et le jeu de données final. Par exemple, si 100 cellules sont visitées dans le jeu de données initial, alors on va comparer le top 10 des cellules initial et anonyme. Supponsons que vous en avec 4 en commun alors vous aurez un meet de 0.4

Cordialement,

BN
